### PR TITLE
Implementing revert opcode

### PR DESCRIPTION
**Pull request by smblucker**
These changes implement the revert opcode as referenced here:

https://github.com/hyperledger/burrow/issues/600

### CODE REVIEW

File: execution/evm/vm.go [hunk 1]
```diff
@@ -210,6 +210,8 @@ func (vm *VM) Call(caller, callee acm.MutableAccount, code, input []byte, value
 				return nil, fmt.Errorf("error transferring value %v %s (callee) -> %s (caller)",
 					value, callee, caller)
 			}
+		} else {
+			childCallState.Sync(callState)
```
Comment (silasdavis) [inline execution/evm/vm.go]: I'm starting to feel `Sync` was a poor choice of name - that said I'm not sure i can think of a better one. Perhaps a comment here would be a nice addition:
```go
// Copy any state updates from child call frame into current call frame
```

File: execution/evm/vm.go [hunk 2]
```diff
@@ -1145,16 +1150,17 @@ func (vm *VM) call(caller acm.Account, callee acm.MutableAccount, code, input []
...
+			callState.UpdateAccount(receiver)
+			callState.RemoveAccount(callee.Address())
 			vm.Debugf(" => (%X) %v\n", addr[:4], callee.Balance())
 			fallthrough
 
 		case STOP: // 0x00
 			return nil, nil
 
-		case STATICCALL, CREATE2:
+		case STATICCALL:
```
Comment (seanyoung) [inline execution/evm/vm.go]: This PR is not changing anything wrt to CREATE2

Comment (smblucker) [reply execution/evm/vm.go]: I made the changes including Silas' request for a comment above.

Comment (smblucker) [reply execution/evm/vm.go]: Some tests are now not passing

**Pull request merged**